lisa/sut_orchestrator/azure/platform_.py

```diff
@@ -687,6 +688,10 @@ def _initialize(self, *args: Any, **kwargs: Any) -> None:
             credential=self.credential, subscription_id=self.subscription_id
         )
 
+        self._rm_client.resource_groups.create_or_update(
```

Comment:
check if it exists before call `create_or_update`. It may be concurrent issues, and it's slow to perform this operation.

Reply:
I am not sure if doing a `exist` check will solve the concurrency issue.  Two threads running in parallel can reach the `exist` check together and try creating the resource group. Maybe we can use locks to sync the threads. Thoughts?

Reply:
If the resource group doesn't exist, it may meet the concurrent issue. But it won't happen, once it exists. But if we call create_or_update each time, it may meet concurrent issue even it exists. If there is high concurrency like 30, it may reach the max deployment script error. So we need to reduce the chance to call create_or_update.

Reply:
Makes sense!